env0/resource_project_policy.go

```diff
@@ -28,39 +25,22 @@ func resourcePolicy() *schema.Resource {
 				Computed:    true,
 			},
 			"project_id": {
-				Type:        schema.TypeString,
-				Description: "id  of the project",
-				Required:    true,
-				ValidateDiagFunc: func(i interface{}, p cty.Path) diag.Diagnostics {
-					if strings.TrimSpace(i.(string)) == "" {
-						return diag.Errorf("project id must not be empty")
-					}
-					return nil
-				},
+				Type:             schema.TypeString,
+				Description:      "id of the project",
+				Required:         true,
+				ValidateDiagFunc: ValidateNotEmptyString,
```

Comment:
Do we really need to have this validation func when attribute is `Required`?

Reply:
So yeah... a weird thing...
It seems `project_id = ""` will path the `required` validation.
I was surprised as well...